### PR TITLE
Feature database models

### PR DESCRIPTION
**Pull request by RRicardotj**
This PR Create migrations and models for a basic checkout

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
     "dev": "nodemon -w src --config --ext js,graphql --exec \"babel-node src\"",
     "build": "npm run cleanup && babel src -D -d dist",
     "start": "node dist/index.js",
-    "cleanup": "rm -rf dist"
+    "cleanup": "rm -rf dist",
+    "db-migrate": "npx sequelize db:migrate",
```
Comment (abelosorio) [inline package.json]: We usually use the semicolon as separator: "db:migrate".

File: package.json [hunk 2]
```diff
@@ -8,7 +8,9 @@
...
+    "db-reset": "npx sequelize db:migrate:undo:all"
```
Comment (abelosorio) [inline package.json]: I don't think we need this script. I'd keep as few scripts as possible in the package.json file. Any script can be run using npx anyway...

File: src/migrations/20200228144719-product.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+'use strict';
```
Comment (abelosorio) [inline src/migrations/20200228144719-product.js]: Let's remove the semicolons here (make sure to run the linter).

File: src/migrations/20200228144719-product.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    const tableFields = {
```
Comment (abelosorio) [inline src/migrations/20200228144719-product.js]: This variable is not needed, you can simply execute the createTable command with the fields data in it.

File: src/migrations/20200228144719-product.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      id: {
+        type: DataTypes.UUID,
+        primaryKey: true,
+        autoIncrement: true,
```
Comment (abelosorio) [inline src/migrations/20200228144719-product.js]: There is not auto increment for UUIDs. Just remove this line.

File: src/migrations/20200228144719-product.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        type: Sequelize.DECIMAL(10, 2),
+        allowNull: false
+      },
+      marketPrice: {
+        type: Sequelize.DECIMAL(10, 2),
+        allowNull: false
+      },
+      createdAt: {
+        allowNull: false,
+        type: Sequelize.DATE,
```
Comment (abelosorio) [inline src/migrations/20200228144719-product.js]: Trailing comma here.

File: src/migrations/20200228144719-product.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      },
+      updatedAt: {
+        allowNull: false,
```
Comment (abelosorio) [inline src/migrations/20200228144719-product.js]: This field can be nullable.

File: src/migrations/20200228144719-product.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        type: Sequelize.DATE,
```
Comment (abelosorio) [inline src/migrations/20200228144719-product.js]: Trailing comma.

File: src/migrations/20200228144719-product.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      }
+    }
+
+    return queryInterface.createTable('product', tableFields)
+  },
+
+  down: (queryInterface) => {
+    return queryInterface.dropTable('product')
+  }
+};
```
Comment (abelosorio) [inline src/migrations/20200228144719-product.js]: Semicolon.

File: src/models/product.js [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  price: {
+    type: DataTypes.DECIMAL(10, 2),
+    allowNull: false
+  },
+  marketPrice: {
+    type: DataTypes.DECIMAL(10, 2),
+    allowNull: false
+  },
+  createdAt: {
```
Comment (abelosorio) [inline src/models/product.js]: These fields are added automatically by Sequelize.

File: package.json (hunk 2 above)
Comment (RRicardotj) [reply package.json]: do you mean both scripts?

Comment (abelosorio) [reply package.json]: Yes!

File: src/migrations/20200228144719-product.js (hunk 3 above)
Comment (RRicardotj) [reply src/migrations/20200228144719-product.js]: Fixed

File: src/migrations/20200228144719-product.js (hunk 4 above)
Comment (RRicardotj) [reply src/migrations/20200228144719-product.js]: Got it!

File: src/models/product.js (hunk 10 above)
Comment (RRicardotj) [reply src/models/product.js]: What fields?

Comment (abelosorio) [reply src/models/product.js]: createdAt and updatedAt

**Pull request merged**